processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -170,9 +167,13 @@ public GroupByMergingQueryRunnerV2(
               final ReferenceCountingResourceHolder<ByteBuffer> mergeBufferHolder;
               try {
                 // This will potentially block if there are no merge buffers left in the pool.
-                final long timeout = timeoutAt - System.currentTimeMillis();
-                if (timeout <= 0 || (mergeBufferHolder = mergeBufferPool.take(timeout)) == null) {
-                  throw new TimeoutException();
+                if (hasTimeout) {
+                  final long timeout = timeoutAt - System.currentTimeMillis();
+                  if (timeout <= 0 || (mergeBufferHolder = mergeBufferPool.take(timeout)) == null) {
+                    throw new TimeoutException();
+                  }
+                } else {
+                  mergeBufferHolder = mergeBufferPool.take(-1);
```

Comment:
In the spirit of blocking APIs from JDK to have a method that takes a timeout and timeUnit, and a method without parameters for indefinite blocking, rather than encoding this by negative argument.